Jenkinsfile

```diff
@@ -88,61 +73,75 @@ pipeline {
+		// the future will be moved to a QA repository, specific to release.
+		stage('Post: Build and run StableIdentifier QA') {
 			steps{
 				script{
-					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]) {
-						sh "mysql -u$user -p$pass -e \'drop database if exists ${env.RELEASE_CURRENT}; create database ${env.RELEASE_CURRENT}\'"
-						sh "mysqldump --opt -u$user -p$pass ${env.SLICE_CURRENT} | mysql -u$user -p$pass ${env.RELEASE_CURRENT}"
+					// Clone data-release-pipeline and checkout specific branch
+					utils.cloneOrUpdateLocalRepo("data-release-pipeline")
+					dir("data-release-pipeline") {
+				        	sh "git checkout feature/post-step-tests-stid-history"
+				        
+						// Build and run QA jar file
+						dir("ortho-stable-id-history") {
+							utils.buildJarFile()
+							withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]) {
+								sh "java -jar target/OrthoStableIdHistory-*-jar-with-dependencies.jar $ConfigFile"
+							}
+						}
 					}
 				}
 			}
 		}
-		// QA for ensuring StableIdentifier instances are proper.
-		stage('Post: StableIdentifier QA'){
+		// This stage creates a new 'release_previous' database from the 'release_current' database,
+		// and takes the recently updated 'slice_current' database and creates a new 'release_current' one.
+		stage('Post: Rotate release DBs'){
 			steps{
 				script{
-					dir("data-release-pipeline/ortho-stable-id-history"){
-						withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')]) {
-							sh "java -jar target/OrthoStableIdHistory-*-jar-with-dependencies.jar $ConfigFile"
-						}
+					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]) {
+						
+						// Take existing 'release_current' DB and replace 'release_previous' with it.
+						// TODO this should come from S3 final DB resting place.
+						def previousReleaseVersion = utils.getPreviousReleaseVersion()
+						def previousReleaseCurrentFinalFilename = "${env.RELEASE_CURRENT_DB}_${previousReleaseVersion}_final.dump"
+						utils.takeDatabaseDump("${env.RELEASE_CURRENT_DB}", "${previousReleaseCurrentFinalFilename}", "${env.RELEASE_SERVER}")
+				      		sh "gzip ${previousReleaseCurrentFinalFilename}"
+						utils.replaceDatabase("${env.RELEASE_PREVIOUS_DB}", "${previousReleaseCurrentFinalFilename}.gz")
+						sh "rm ${previousReleaseCurrentFinalFilename}.gz"
+						
+						// Replace 'release_current' DB with updated 'slice_current' DB.
+				    		def releaseVersion = utils.getReleaseVersion()
+						def sliceCurrentDump = "${env.SLICE_CURRENT_DB}_${releaseVersion}.dump"
+						def databaseDumpFilename = utils.takeDatabaseDumpAndGzip("${env.SLICE_CURRENT_DB}", "update_stable_ids", "after", "${env.RELEASE_SERVER}")
+			            		sh "echo ${databaseDumpFilename}"
```

Comment:
Just curious why the name is captured in a variable ane echoed?

Reply:
Hmm, probably a typo

Reply:
Old debugging statement?

Reply:
Removed it, thanks for catching that